tests/TestableIO.System.IO.Abstractions.TestingHelpers.Tests/MockDirectoryTests.cs

```diff
@@ -813,6 +814,104 @@ public void MockDirectory_CreateDirectory_ShouldSucceedIfTryingToCreateUNCPathSh
             Assert.IsTrue(fileSystem.Directory.Exists(@"\\server\share\"));
         }
 
+#if FEATURE_CREATE_TEMP_SUBDIRECTORY
+        [Test]
+        public void MockDirectory_CreateTempSubdirectory_ShouldCreateSubdirectoryInTempDirectory()
+        {
+            // Arrange
+            var fileSystem = new MockFileSystem();
+
+            // Act
+            var result = fileSystem.Directory.CreateTempSubdirectory();
+
+            // Assert
+            Assert.IsTrue(fileSystem.Directory.Exists(result.FullName));
+            Assert.IsTrue(result.FullName.StartsWith(Path.GetTempPath()));
+        }
+
+        [Test]
+        public void MockDirectory_CreateTempSubdirectory_ShouldHaveCharactersFromUpperLowerOrDigits()
+        {
+            // Arrange
+            var fileSystem = new MockFileSystem();
+
+            // Act
+            var result = fileSystem.Directory.CreateTempSubdirectory();
+
+            // Assert
+            var name = Path.GetFileName(result.FullName);
+            for (var i = 0; i < name.Length; i++)
+            {
+                var c = name[i];
+                if (!char.IsUpper(c) && !char.IsLower(c) && !char.IsDigit(c))
```

Comment:
This check is incorrect (except on .NET Framework)!
I ran these tests an the real file system ([see here](https://github.com/Testably/Testably.Abstractions/pull/286)) and the result is e.g. `hnkjxa50.lqs`, so it contains a dot as well!

Reply:
That's interesting. I'm not sure how that was failing on the Windows platform as the name generation shouldn't have been generating `.` characters. The implementation has changed now, and this test was removed due to reliance on the `Path.GetRandomFileName()`.